### PR TITLE
docs: Render mermaid graphs

### PR DESCRIPTION
**Pull request by lewismiddleton**
Add the plugin as a dependency as per [installation guide](https://github.com/fralau/mkdocs-mermaid2-plugin#installation).
`poetry add mkdocs-mermaid2-plugin --group docs`

Add to the `mkdocs.yml` via [plugins syntax](https://www.mkdocs.org/dev-guide/plugins/)

This on it's own is sufficient to get the diagrams to render.

Resolves: #567

### CODE REVIEW

**Comment by lewismiddleton** [comment]
Adding a `fence_mermaid_custom` format to superfences as per [mermaid2 plugin Material theme config](https://github.com/fralau/mkdocs-mermaid2-plugin#additional-settings-for-the-material-theme) didn't seem to change behaviour but [seems like is best practice](https://github.com/fralau/mkdocs-mermaid2-plugin/issues/77#issuecomment-1642100370) if you want the diagrams to inherit styling from any theming you choose.

```
plugins: 
  - mermaid2

markdown_extensions:
  - pymdownx.superfences:
      custom_fences:
        - name: mermaid
          class: mermaid
          format: !!python/name:mermaid2.fence_mermaid_custom
```

File: mkdocs.yml [hunk 1]
```diff
@@ -56,7 +59,11 @@ markdown_extensions:
       emoji_generator: !!python/name:materialx.emoji.to_svg
   - pymdownx.inlinehilite
   - pymdownx.snippets
-  - pymdownx.superfences
+  - pymdownx.superfences:
+      custom_fences:
+        - name: mermaid
+          class: mermaid
+          format: !!python/name:mermaid2.fence_mermaid_custom
```
Comment (tcarmet) [inline mkdocs.yml:66]: Hey @lewismiddleton! Thanks for your contribution and for spotting this issue, very much appreciated! 

In my other mkdocs project I did the following:
```suggestion
  - pymdownx.superfences:
      custom_fences:
        - name: mermaid
          class: mermaid
          format: !!python/name:pymdownx.superfences.fence_code_format
```

For me it seems like a healthier change as it doesn't require the extra dependency to be installed, would you mind giving it a shot see if it renders as you would like compared with `mermaid2`.

Thanks in advance!

Comment (lewismiddleton) [reply mkdocs.yml:66]: Yeah I agree, a better solution if it works. I wasn't able to get that working, give it a try on your environment and see what you think?

You're right though in that [the docs](https://squidfunk.github.io/mkdocs-material/reference/diagrams/#configuration) say it should be sufficient. I'm curious to know what the difference is between this project and others (if there is actually an issue - idk maybe my environment is just differnet)

Comment (tcarmet) [reply mkdocs.yml:66]: I did try it and it worked.

Worth mentioning though that I was running my doc website through a `mkdocs serve` and reaching it on `localhost:8000` through my browser and not via the `mkdocs build`, it seems like you were using the build method and I don't know if that may cause some differences in the rendering.

Regardless I have another website with those options hosted on GitHub pages that is working fine as well for rendering the mermaid graphs.

Let's give it a go with only the suggestion above, merge the PR, then if it's doesn't work we'll review the addition of the plugin.

Comment (lewismiddleton) [reply mkdocs.yml]: Yeah I was using `poetry run mkdocs build --strict` but can confirm `mkdocs serve` works on my machine too.

I've reverted the commits and pushed the suggested change.

**Review by tcarmet — APPROVED** [review]

File: mkdocs.yml (hunk 1 above)
Comment (tcarmet) [reply mkdocs.yml]: Awesome! Thank you for your patience and making the suggested changes ❤️ 

And good to know it doesn't behave the same with a `mkdocs build`!

**Pull request merged**